Lib/asyncio/timeouts.py

```diff
@@ -125,6 +131,16 @@ def _on_timeout(self) -> None:
         # drop the reference early
         self._timeout_handler = None
 
+    @staticmethod
+    def _insert_timeout_error(exc_val: BaseException) -> None:
+        while exc_val.__context__ is not None:
+            if type(exc_val.__context__) is exceptions.CancelledError:
+                te = TimeoutError()
+                te.__context__ = te.__cause__ = exc_val.__context__
```

Comment:
Why context *and* cause? Can't we decide which to use? Or is that tradition for chaining?

Reply:
Because the current code
```py
                raise TimeoutError from exc_val
```
sets both `__context__` and `__cause__`.

Also, when StopIteration or StopAsyncIteration are raised in a generator, they are set as `__context__` and `__cause__` of a RuntimeError.

Reply:
Okay, got it. The cause/context distinction is too subtle. :-(